### PR TITLE
Schreier sims hpp

### PR DESCRIPTION
**Pull request by Joseph-Edwards**
This PR updates `SchreierSims` in preparation for the v3 release

### CODE REVIEW

**Comment by Joseph-Edwards** [comment]
This will be a draft until the Python bindings are added.

File: include/libsemigroups/schreier-sims.hpp [hunk 1]
```diff
@@ -749,8 +749,10 @@ namespace libsemigroups {
     // isn't true, because we either run to the end or haven't done anything at
     // all.
     // not noexcept because sift_no_checks isn't
-    [[nodiscard]] bool const_contains(const_element_reference x) const;
```
Comment (james-d-mitchell) [inline include/libsemigroups/schreier-sims.hpp:545]: Should be `currently_contains` for consistency! @Joseph-Edwards

**Comment by Joseph-Edwards** [comment]
For some reason, [cygwin failed](https://github.com/libsemigroups/libsemigroups/actions/runs/11900207756/job/33160574327?pr=617) after a recent push. I haven't been able to reproduce this locally, so hopefully the magic of "running the tests again" will fix this. Might be something to do with running out of memory? The object in the test seems quite big.

**Comment by Joseph-Edwards** [comment]
The magic of running the tests again unfortunately didn't fix the problem.

**Comment by Joseph-Edwards** [comment]
Apart from the formatting (which I will fix when I'm not booted in Windows), all tests are now passing.

The cygwin issue was related to the new test we added to check the copy and move operators/constructors. The `SchreierSims` object we constructed was quite large and I think this was causing a stack overflow. Before fixing the issue, I adjusted the stack size of the test executable in cygwin and the tests passed, so hopefully this means there isn't anything else fishy going on.

To fix the test, I worked with pointers instead. @james-d-mitchell, would you mind taking a look at the changes I made in 72f8bee62e13e53d19dfbda0c536ec21a5686f72 to make sure that the test is now checking what we want it to check please? I'm not sure if I'm dereferencing in the right places.

**Comment by james-d-mitchell** [comment]
> To fix the test, I worked with pointers instead. @james-d-mitchell, would you mind taking a look at the changes I made in [72f8bee](https://github.com/libsemigroups/libsemigroups/commit/72f8bee62e13e53d19dfbda0c536ec21a5686f72) to make sure that the test is now checking what we want it to check please? I'm not sure if I'm dereferencing in the right places.

Looks good to me, thanks for finding the fix for this!

**Comment by james-d-mitchell** [comment]
@Joseph-Edwards is there anything else you wanted to do before review?

**Comment by Joseph-Edwards** [comment]
> @Joseph-Edwards is there anything else you wanted to do before review?

No, this is good to go on my end.

**Comment by james-d-mitchell** [comment]
Looks good, thanks @Joseph-Edwards

**Pull request merged**